lua/packer.lua

```diff
@@ -178,8 +180,17 @@ packer.install = function(...)
       await(a.main)
       plugin_utils.update_helptags(install_paths)
       plugin_utils.update_rplugins()
-      local delta = string.gsub(vim.fn.reltimestr(vim.fn.reltime(start_time)), ' ', '')
-      display_win:final_results(results, delta)
+
+      if not vim.tbl_isempty(results.rocks) then
+        await(neorocks.install_rocks(results.rocks))
+      else
+        log.info("No rocks to install")
+      end
+
+      vim.schedule_wrap(function()
```

Comment:
Rather than placing this in a `schedule_wrap` call, let's move the `await(a.main)` line down after all the async code, to keep things cleaner. This will also entail moving the helptags and rplugins updates, since they too need to be on the main thread.

Reply:
hmmm I'm not sure exactly what you mean.

Reply:
Ah - `await(a.main)` (on line 180) is a little hack to ensure that the code after that line until the next `await` call is running on the main event loop (which I sometimes incorrectly call the "main thread"). It's currently used to make sure that updating the helptags and remote plugins happens on the main event loop. Here, I'm suggesting that we could move the async code that currently runs *after* the helptags, etc. (which is the neorocks install call) to *before* the call to `await(a.main)`, which would mean that the code you have wrapped with `vim.schedule_wrap` here no longer needs to be wrapped (because there will no longer be any async code between it and the last `await(a.main)` call). That should make this cleaner and very slightly more efficient (no need to wait for the main event loop twice).

Reply:
Hm, I'm still not sure I 100% understand what you're thinking for where to place it and how tht works. I'm fine w/ either you adding a commit to the branch or we just do it after (if it's not a big deal)

Reply:
Sure - let's get V1 ready except for this, then I can add a commit before merging.